pkg/clients/aws/eks/eks_test.go

```diff
@@ -0,0 +1,184 @@
+package eks
+
+import (
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/onsi/gomega"
+)
+
+// MockAMIClient mocks AMI client which is used to get information about AMI images
+type MockAMIClient struct {
+	MockDescribeImages func(*ec2.DescribeImagesInput) (*ec2.DescribeImagesOutput, error)
+}
+
+// DescribeImages Describes AMI images for the given input
+func (m *MockAMIClient) DescribeImages(input *ec2.DescribeImagesInput) (*ec2.DescribeImagesOutput, error) {
+	return m.MockDescribeImages(input)
+}
+
+var mockImages = []*ec2.Image{
+	{
+		CreationDate: aws.String("2019-08-13T11:38:33.006Z"),
+		ImageId:      aws.String("img0"),
+	},
+	{
+		CreationDate: aws.String("2019-08-14T11:38:33.001Z"),
+		ImageId:      aws.String("img1"),
+	},
+	{
+		CreationDate: aws.String("2019-08-14T11:38:33.000Z"),
+		ImageId:      aws.String("img2"),
+	},
+}
+
+func TestMain(m *testing.M) {
+	os.Exit(m.Run())
+}
+
+func TestGetMostRecent(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	cases := []struct {
+		expected string
+	}{
+		// img1 is the most recent image
+		{"img1"},
```

Comment:
consider testing getMostRecent with more than 1 image in the list

Reply:
the `"img1"` is not the set, it's the expected value. This is testing against `mockImages` list which has 3 items

Reply:
the used list here (`mockImages`) has 3 instances